packages/workers/src/build-latest-cache.ts

```diff
@@ -0,0 +1,200 @@
+    db.exec(
+      Array.from(watchContractListCheckSum)
+        .map(
+          (address) =>
+            `INSERT OR REPLACE INTO contract_start_block (contract_address, first_block_number) VALUES ('${address}', ${startBlockOverride});`,
+        )
+        .join('\n'),
+    )
+
+    db.prepare(
+      'INSERT OR REPLACE INTO latest_block_processed (id, latest_block_processed) VALUES (1, ?)',
+    ).run(startBlockOverride)
+  } else {
+    const lastProcessedBlockNumber = await getStartBlockNumberFn(db, provider)
+    db.exec(
+      Array.from(watchContractListCheckSum)
+        .map(
+          (address) =>
+            `INSERT OR IGNORE INTO contract_start_block (contract_address, first_block_number) VALUES ('${address}', ${lastProcessedBlockNumber});`,
+        )
+        .join('\n'),
+    )
+  }
+
+  const indexer = new BlockRunner({
+    provider,
+    chainId,
+    getStartBlockNumberFn: () => getStartBlockNumberFn(db, provider),
+    processBlockFn: async (blockNumber) =>
+      processBlockFn({
+        provider,
+        blockNumber,
+        watchContractListLowercase: new Set(
+          Array.from(watchContractListCheckSum).map((address) =>
+            address.toLowerCase(),
+          ),
+        ),
+        db,
+      }),
+    onError: async (latestSafeProcessedBlock: number) =>
+      onError(latestSafeProcessedBlock, db),
+  })
+
+  await indexer.start()
+}
+
+async function getDeFiContractAddressesCheckSum({
+  defiProvider,
+  chainId,
+  chainName,
+}: { defiProvider: DefiProvider; chainId: Chain; chainName: string }) {
+  const defiPoolAddresses = await defiProvider.getSupport({
+    filterChainIds: [chainId],
+  })
+
+  const watchContractList = new Set<string>()
+  for (const pools of Object.values(defiPoolAddresses || {})) {
+    for (const pool of pools) {
+      for (const address of pool.protocolTokenAddresses?.[chainId] || []) {
+        watchContractList.add(getAddress(address).slice(2))
+      }
+    }
+  }
+
+  logger.info(
+    `Watching ${watchContractList.size} DeFi contracts on ${chainName}`,
+  )
+  return watchContractList
+}
+
+async function getStartBlockNumberFn(
+  db: DatabaseType,
+  provider: CustomJsonRpcProvider,
+): Promise<number> {
+  const lastPrcessedBlock = (
+    db
+      .prepare('SELECT latest_block_processed FROM latest_block_processed')
+      .get() as { latest_block_processed?: number } | undefined
+  )?.latest_block_processed
+
+  if (lastPrcessedBlock) {
+    return lastPrcessedBlock
+  }
+
+  const latestBlockNumber = await provider.getBlockNumber()
+
+  return latestBlockNumber
+}
+
+async function processBlockFn({
+  provider,
+  blockNumber,
+  watchContractListLowercase,
+  db,
+}: {
+  provider: CustomJsonRpcProvider
+  blockNumber: number
+  watchContractListLowercase: Set<string>
+  db: DatabaseType
+}): Promise<void> {
+  const receipts = await provider.send('eth_getBlockReceipts', [
+    `0x${ethers.toBeHex(blockNumber).slice(2).replace(/^0+/, '')}`,
+  ])
```

Comment:
I've just tested `eth_getBlockReceipts` with multiple leading 0s and the result is correct.

```
{
    "id": 1,
    "jsonrpc": "2.0",
    "method": "eth_getBlockReceipts",
    "params": ["0x00001324bf8"]
}
```

Why are we removing the leading 0s from the number here?

Also, does `ethers.toBeHex` even add leading 0s?